### DISCUSSION TITLE
A command does not work

### DISCUSSION THREAD

**Discussion by Charlie-techLab** [microsoft/RAG_Hack#66] [Show and tell]
Hello!

I want to share with you a finding. When I was following the instructions provided here: https://github.com/Azure-Samples/azure-search-openai-demo/

![image](https://github.com/user-attachments/assets/1174a99b-2fbf-4d05-ad44-34885f781d0a)

the command  `azd auth login` did not work 
![image](https://github.com/user-attachments/assets/88a11904-cf74-453e-b5ac-3b6b55c5badf)

but when I used this one: az login --use-device-code I could enter to the Azure Portal. I hope you find it useful in some way if you encounter with the same issue.

**Comment by pamelafox** [comment]
Was that in Codespaces? It's supposed to autodetect that you need the device code flow, but sometimes we hear that it isn't doing that correctly. In that case, you should use:

azd auth login --use-device-code

**Reply by Charlie-techLab** [reply]
Hi!

Yes, it was in Codespaces. 

Thank you very much for providing this information. 

Indeed, when I was using the `az login --use-device-code` command, the `az up`  command failed

**Reply by pamelafox** [reply]
Yes, that's because az and azd are two different CLIs. I've updated our README now to suggest "azd auth login --use-device-code" when standard login doesn't work. Thanks!

**Reply by Charlie-techLab** [reply]
Likewise, thank you!

**Reply by Charlie-techLab** [reply]
Hello!

I want to follow this guide using CodeSpaces: https://github.com/Azure-Samples/rag-postgres-openai-python/

but neither the `azd auth login` command  nor this one `az login --use-device-code` works, as you can see:

![image](https://github.com/user-attachments/assets/a2f88cb3-bd24-4fad-b177-914b0ef9f7a3)

![image](https://github.com/user-attachments/assets/91411bb2-f805-4ace-bf1d-87a8c0b85da4)


Please, can you give me some guidance on this issue?

Thanks!

**Reply by pamelafox** [reply]
Please try:

 azd auth login --use-device-code 

I'll add that to the rag-postgres repo as well.

**Reply by Charlie-techLab** [reply]
Ok, thank you. I am going to try it now

**Reply by Charlie-techLab** [reply]
Now, I can confirm it works: ![image](https://github.com/user-attachments/assets/fd065d45-2514-4b9d-81a8-3444a8de38f5)
Thank you!

**Comment by Charlie-techLab** [comment]
Hello 

Also here it would be useful to update within Codespaces: https://github.com/Azure-Samples/Cosmic-Food-RAG-app since only with this command:  `azd auth login --use-device-code` it is permitted to log in into Azure

**Reply by Charlie-techLab** [reply]
Hello!

I have followed this template: https://github.com/Azure-Samples/Cosmic-Food-RAG-app
but I get this: https://github.com/user-attachments/assets/7fb557dc-8a8a-4684-8550-5330e7a1eb46

Please, can you give me some guidance on this issue?

Thanks!

**Reply by pamelafox** [reply]
That sort of error indicates an error in your server typically. I recommend checking your server logs to see what's happened. It's usually easier to check them when running locally versus on the deployed version.

cc @john0isaac who maintains that project

**Reply by john0isaac** [reply]
@Charlie-techLab just like Pamela said this is an error coming from the server.
Most likely the response of the request is not in the desired format.
So, the frontend is not able to parse it.
Without the logs I can't tell you why this is happening but most probable causes are: 
- there is an actual error that happened during the deployment of the resources and you will need to re run azd up to fix it.
Or
- you didn't populate the database with data.. we don't automatically populate the database with the provided sample of data this step has to be done manually after the deployment is done. See this https://github.com/Azure-Samples/Cosmic-Food-RAG-app#add-the-data


Can you inspect the browser and take a screenshot of the response that is being returned from the backend? that would really help.
https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/network/#open-the-network-tool

**Reply by Charlie-techLab** [reply]
Hi @john0isaac 

Recently, I had an issue with my Azure subscription and I think that is why I am having this issue. I am waiting for response. 

Thank you very much for being attentive!
